bayesian/api_v1.py

```diff
@@ -1242,18 +1242,48 @@ class CveByDateEcosystem(Resource):
     method_decorators = [login_required]
 
     @staticmethod
-    def get(modified_date, ecosystem=None):
+    def get(end_date, ecosystem='all'):
         """Implement GET Method."""
-        if not modified_date:
+        if not end_date:
             raise HTTPError(400, error="Expected date in the request")
         try:
-            datetime.datetime.strptime(modified_date, '%Y%m%d')
+            datetime.datetime.strptime(end_date, '%Y%m%d')
         except ValueError:
             msg = 'Invalid datetime specified. Please specify in YYYYMMDD format'
             raise HTTPError(400, msg)
-        getcve = CveByDateEcosystemUtils(modified_date, ecosystem)
-        result = getcve.get_cves_by_date() if not ecosystem else getcve.get_cves_by_date_ecosystem()
-        return jsonify(result), 200
+        try:
+            cve_sources = request.args.get('cve_sources', 'all')
+            date_range = int(request.args.get('date_range', 7))
+            if date_range <= 0:
+                raise HTTPError(400, error="date_range parameter should be a positive integer")
+            getcve = CveByDateEcosystemUtils(None, cve_sources,
+                                             end_date, ecosystem, date_range)
+            result = getcve.get_cves_by_date_ecosystem()
+        except Exception as e:
+            result = {}
+            current_app.logger.error('ERROR: {}'.format(str(e)))
+            raise HTTPError()
```

Comment:
You have not followed it properly https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/bayesian/api_v1.py#L300
This HTTPError() would fail on its own.
This tells me that exception is not tested by test cases. Please add valid unit test cases to test the exception as well.